TestDispatchFromFlagsAndUseBuiltInArgs fails on arm

https://github.com/openshift/imagebuilder/pull/224  introduces a test that fail on arm64, armhf and armel with:

On arm64:
```
=== RUN   TestDispatchFromFlagsAndUseBuiltInArgs
    dispatchers_test.go:798: Expected linux/arm64, to match linux/arm64/v8
--- FAIL: TestDispatchFromFlagsAndUseBuiltInArgs (0.00s)
```

on armel:
```
=== RUN   TestDispatchFromFlagsAndUseBuiltInArgs
    dispatchers_test.go:798: Expected linux/arm, to match linux/arm/v8
--- FAIL: TestDispatchFromFlagsAndUseBuiltInArgs (0.00s)
```

On armhf:

```
=== RUN   TestDispatchFromFlagsAndUseBuiltInArgs
    dispatchers_test.go:798: Expected linux/arm, to match linux/arm/v7
--- FAIL: TestDispatchFromFlagsAndUseBuiltInArgs (0.00s)
```

--- a/dispatchers_test.go
+++ b/dispatchers_test.go
@@ -931,6 +931,9 @@ func TestDispatchFromFlags(t *testing.T) {
 
 func TestDispatchFromFlagsAndUseBuiltInArgs(t *testing.T) {
 	expectedPlatform := localspec.OS + "/" + localspec.Architecture
+	if localspec.Variant != "" {
+		expectedPlatform += "/" + localspec.Variant
+	}
 	mybuilder := Builder{
 		RunConfig: docker.Config{
 			WorkingDir: "/root",

